src/hmem_rocr.c

```diff
@@ -1054,6 +1060,48 @@ int rocr_dev_reg_copy_from_hmem(uint64_t handle, void *dest, const void *src,
 	return FI_SUCCESS;
 }
 
+bool rocr_is_dmabuf_supported(void)
+{
+	hsa_status_t hsa_ret;
+	bool dmabuf_support = false, dmabuf_export = false;
+	res = hsa_ops.hsa_system_get_info((hsa_system_info_t) 0x204, &dmabuf_support);
+
+	if (res != HSA_STATUS_SUCCESS)
+	{
+		FI_WARN(&core_prov, FI_LOG_CORE,
+			"Failed to retrieve system info (dmabuf): %s\n",
+			ofi_hsa_status_to_string(hsa_ret));
+		return -FI_EIO;
+	}
+
+#if HAVE_HSA_AMD_PORTABLE_EXPORT_DMABUF
+	dmabuf_export = true;
+#endif
+
+	return dmabuf_support && dmabuf_export;
+}
+
+int rocr_hmem_get_dmabuf_fd(void *addr, uint64_t size, int *dmabuf_fd, 
+			    uint64_t *offset)
+{
```

Comment:
i missed the macro guard here for dmabuf_support, thought `rocr_is_dmabuf_supported()` would suffice, but forgot about compilation